counter_block.py

```diff
@@ -143,17 +143,13 @@ def _calculate_next(self, now):
 
         return date
 
-    def stop(self):
-        self._store()
-        self._backup()
-
     def process_signals(self, signals):
         signals_to_notify = []
         self.for_each_group(self.process_group, signals,
                             kwargs={"to_notify": signals_to_notify})
 
         self.notify_signals(signals_to_notify)
-        self._store()
+        self._save()
```

Comment:
I'm not sure we should be saving the persistence every time a signal comes through. The point of the mixin is to periodically save on a backup interval (in case of emergency) but then to save when the block shuts down properly. Putting a `save` here will have some noticeable impact on the block's performance I think since before we were just setting the values to the dictionary, not actually writing to disk.